File: algorithm/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.go
import "sort"

func solution(sequence []int, k int) []int {
    if sequence[0] == k {
        return []int{0, 0}
    }
    
    candidates := getCandidates(sequence, k)
    sort.Slice(candidates, func(i, j int) bool {
        return candidates[i][1] - candidates[i][0] < candidates[j][1] - candidates[j][0]
    })
    
    return candidates[0]
}

func getCandidates(sequence []int, k int) [][]int {
    candidates := [][]int{}
    
    start, end := 0, 1
    subSeqSum := sequence[0] + sequence[1]
    for start <= end {
        if subSeqSum == k {
            candidates = append(candidates, []int{start, end})
        }
        
        if subSeqSum > k || end == len(sequence) - 1 {
            subSeqSum -= sequence[start]
            start += 1
        } else {
            end += 1
            subSeqSum += sequence[end]
        }
    }
    
    return candidates
}

File: algorithm/프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    if sequence[0] == k:
        return [0, 0]
    
    candidates = []
    
    start, end = 0, 1
    subSeqSum = sequence[0] + sequence[1]
    while start <= end:
        if subSeqSum == k:
            candidates.append([start, end])
        
        if subSeqSum > k or end == len(sequence) - 1:
            subSeqSum -= sequence[start]
            start += 1
        else:
            end += 1
            subSeqSum += sequence[end]
    
    sortedCandidates = sorted(candidates, key = lambda x : x[1] - x[0])
    
    return sortedCandidates[0]